system/Database/Postgre/Connection.php

```diff
@@ -69,8 +69,13 @@ public function connect(bool $persistent = false)
             $this->DSN = mb_substr($this->DSN, 6);
         }
 
-        // Convert semicolons to spaces.
-        $this->DSN = str_replace(';', ' ', $this->DSN);
+        // Convert semicolons to spaces in DSN format like:
+        // pgsql:host=localhost;port=5432;dbname=database_name
+        $this->DSN = str_replace(
+            [';port=', ';user=', ';password=', ';dbname='],
+            [' port=', ' user=', ' password=', ' dbname='],
+            $this->DSN
+        );
```

Comment:
If a password contains string `;port=`, it cannot connect to the database.

Reply:
Will try to fix this.